components/styled/button/src/atoms/button.tsx

```diff
@@ -120,7 +131,8 @@ const StyledButton = styled.button<StyledProps>`
     width: auto;
   }
 
-  ${({ buttonStyle }): SerializedStyles => setColors(buttonStyle)}
+  ${({ padding }): SerializedStyles => setButtonPadding(padding)}
+  ${({ buttonStyle }): SerializedStyles => setColors(buttonStyle ?? 'primary')}
```

Comment:
unnecessary duplication of the default between this and line 149, idm which you keep though